packages/teleterm/src/ui/services/workspacesService/workspacesService.ts

```diff
@@ -134,17 +134,26 @@ export class WorkspacesService extends ImmutableStore<WorkspacesState> {
       });
     };
 
-    const isConnected = this.clustersService.findCluster(clusterUri)?.connected;
-    return new Promise((resolve, reject) => {
-      if (clusterUri && !isConnected) {
-        this.modalsService.openClusterConnectDialog(clusterUri, () => {
-          setWorkspace();
-          resolve();
-        });
-      } else {
+    const cluster = this.clustersService.findCluster(clusterUri);
+    if (!cluster) {
+      return Promise.resolve();
+    }
```

Comment:
Which scenario does this code cover? It seems like it's going to behave differently now if the cluster wasn't found?

Reply:
One or two times I had a situation where cluster wasn't found and our code tried to operate on that `undefined` cluster which resulted in error (we were showing login dialog and that failed with error `cannot read name (clusterName) of undefined`). 
The worst thing is that I can't remember why it happened :( 
I added a notification and logging, so we will be able to track this problem.